### PR TITLE
[Merged by Bors] - Fix gracefull shutdown

### PR DESCRIPTION
**Pull request by almogdepaz**
## Motivation
process gracefull shutdown using intruppt during sync dosen't work properly
causes users to have to kill the node

### CODE REVIEW

File: sync/fetch_queue.go [hunk 1]
```diff
@@ -48,8 +48,16 @@ func concatShortIds(items []types.Hash32) string {
 	return str
 }
 
+func (fq *fetchQueue) shutdownRecover() {
```
Comment (almogdepaz) [inline sync/fetch_queue.go:51]: the purpose of this hack is to hide the stack trace of the panic we currently might get due to flaws in the gracefull shutdown process.

File: sync/syncer.go [hunk 2]
```diff
@@ -516,13 +515,13 @@ func (s *Syncer) validateBlockView(blk *types.Block) bool {
 
 func validateVotes(blk *types.Block, forBlockfunc ForBlockInView, depth int, lg log.Log) (bool, error) {
 	view := map[types.BlockID]struct{}{}
-	for _, blk := range blk.ViewEdges {
-		view[blk] = struct{}{}
+	for _, b := range blk.ViewEdges {
```
Comment (almogdepaz) [inline sync/syncer.go:537]: blk was shadowed...

**Review by noamnelke — APPROVED** [review]
I think this isn't a great design and we should probably implement a better shutdown mechanism, but it quickly solves an annoyance in the short term and it's not worse than the existing code, so I see no reason not to merge this.

**Comment by almogdepaz** [comment]
bors merge

**Pull request closed without merging**